examples/test_create_model_api.py

```diff
@@ -0,0 +1,51 @@
+# %%
+import subprocess
+from pathlib import Path
+
+import imod
+import numpy as np
+from hydromt.log import setuplog
+
+from hydromt_imod.imod import ImodModel
+
+logger = setuplog("Initialize logger", log_level=10)
+
+root = Path(r"c:\path\to\root")
+config_fn = root / r"example_imod.ini"
+
+
+# %%
+
+mod = ImodModel(root=root, mode="w", data_libs="artifact_data", logger=logger)
+
+# Configure
+mod.read_config(config_fn=config_fn)
+
+# Tests on setup_staticmaps_from_raster
+# bbox in EPSG:4326
+bbox = np.array([11.37583385, 45.19166655, 13.08333353, 46.85166659])
+
+mod.setup(region={"bbox": bbox}, res=1000.0)
```

Comment:
This follows a little the previous comment but your model does not need a general setup method. This is done by the functions `build` and `update` of the core. They use the dict of hydromt ini file to run the setup_ methods in the order they are listed in the ini file. And in case of build starting with the method that takes care of the region argument in your case setup_grid.

As of hydromt 0.7 release, ``res`` is not an argument of `build` anymore and should be passed to the setup_grid method directly (eg in your config file example_imod.ini).
See https://deltares.github.io/hydromt/latest/_generated/hydromt.Model.build.html#hydromt.Model.build

```suggestion
mod.build(region={"bbox": bbox}, opt=opt)
```